wgpu/src/backend/webgpu.rs

```diff
@@ -1315,9 +1356,11 @@ impl crate::context::Context for ContextWebGpu {
         let mut mapped_formats = formats.iter().map(|format| map_texture_format(*format));
         // Preferred canvas format will only be either "rgba8unorm" or "bgra8unorm".
         // https://www.w3.org/TR/webgpu/#dom-gpu-getpreferredcanvasformat
-        let preferred_format = self.0.get_preferred_canvas_format();
-        if let Some(index) = mapped_formats.position(|format| format == preferred_format) {
-            formats.swap(0, index);
+        if let Some(gpu) = &self.gpu {
```

Comment:
I think I could panic here instead. It shouldn't be possible to get here since `request_adapter` would return `None` if `self.gpu == None`. Should I do that instead?

Reply:
an `assert` on the definedness would make sense, that also allows then to document why we think that it can't be undefined at this point